src/Dfc.CourseDirectory.Functions/Startup.cs

```diff
@@ -28,12 +28,13 @@ public override void Configure(IFunctionsHostBuilder builder)
             builder.Services.AddSingleton(configuration);
 
             builder.Services.AddSqlDataStore(configuration.GetConnectionString("DefaultConnection"));
-			
+
             builder.Services.AddCosmosDbDataStore(
                 endpoint: new Uri(configuration["CosmosDbSettings:EndpointUri"]),
                 key: configuration["CosmosDbSettings:PrimaryKey"]);
 
             builder.Services.AddTransient<LarsDataImporter>();
+            builder.Services.AddTransient<IUkrlpWcfClientFactory, UkrlpWcfClientFactory>();
```

Comment:
Singleton?

Reply:
It's a trivial class so I figured the cost of instantiating it was near zero, and using `Transient` makes it clear there's no shared state. Is there a reason to prefer `Singleton` that I'm unaware of?

Reply:
I tend to use `Singleton` where I can, even in trivial places. (Lots of trivial performance hits = large cumulative performance hit)

Reply:
I remain unconvinced that it should be changed but appreciate the input.